app/logic/events.py

```diff
@@ -139,17 +139,29 @@ def getStudentLedEvents(term):
 
     return programs
 
-def getTrainingEvents(term):
+def getTrainingEvents(term, user):
     """
         The allTrainingsEvent query is designed to select and count eventId's after grouping them
         together by id's of similiar value. The query will then return the event that is associated
         with the most programs (highest count) by doing this we can ensure that the event being
         returned is the All Trainings Event.
+        term: expected to be the ID of a term
+        user: expected to be the current user
+        return: a list of all trainings the user can view
     """
-    trainingEvents = (Event.select(Event)
-                           .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
-                           .where(Event.isTraining,
-                                  Event.term == term))
+    if (user.isAdmin) or (user.isStudent and user.isBonnerScholar):
+        trainingEvents = (Event.select(Event)
+                               .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
+                               .where(Event.isTraining,
+                                      Event.term == term))
+    else:
+        trainingEvents = (Event.select(Event)
+                                .join(ProgramEvent)
+                                .join(Program)
+                                .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
+                                .where(Event.isTraining,
+                                        Event.term == term,
+                                        Program.isBonnerScholars == False))
```

Comment:
This will return duplicate all volunteer trainings, since it is part of a bunch of programs

Reply:
Also we should combine these queries. The only real difference is that we add an extra where expression in the second case.